x/superfluid/keeper/grpc_query.go

```diff
@@ -392,6 +392,46 @@ func (q Querier) EstimateSuperfluidDelegatedAmountByValidatorDenom(goCtx context
 	}, nil
 }
 
+func (q Querier) TotalDelegationByValidatorForDenom(goCtx context.Context, req *types.QueryTotalDelegationByValidatorForDenomRequest) (*types.QueryTotalDelegationByValidatorForDenomResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	var delegationsByValidator = []*types.Delegations{}
```

Comment:
So I figured using pointers here is awkward, which made me follow into the rabbit hole until I noticed that the response struct itself is taking pointer referenced array. 

```
type QueryTotalDelegationByValidatorForDenomResponse struct {
	AssetResponse []*Delegations `protobuf:"bytes,1,rep,name=asset_response,json=assetResponse,proto3" json:"asset_response,omitempty"`
}
```

whilst I think it should instead be 

```

type QueryTotalDelegationByValidatorForDenomResponse struct {
	AssetResponse []Delegations `protobuf:"bytes,1,rep,name=asset_response,json=assetResponse,proto3" json:"asset_response,omitempty"`
}
```
just like other proto response structs do. I do think that the possible solution for this is adding `[ (gogoproto.nullable) = false ];` to the proto field. Please lmk if it works out

Reply:
We should definitely do this ^

Reply:
Could we address this before merging? Thanks!